references/cli/addon.go

```diff
@@ -357,7 +358,64 @@ func NewAddonStatusCommand(c common.Args, ioStream cmdutil.IOStreams) *cobra.Com
 			return nil
 		},
 	}
-	cmd.Flags().BoolVarP(&verboseSatatus, "verbose", "v", false, "show addon descriptions and parameters in addition to status")
+	cmd.Flags().BoolVarP(&verboseStatus, "verbose", "v", false, "show addon descriptions and parameters in addition to status")
+	return cmd
+}
+
+// NewAddonCreateCommand creates an addon scaffold
+func NewAddonCreateCommand() *cobra.Command {
+	var helmRepoURL string
+	var chartName string
+	var chartVersion string
+	var path string
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "bootstrap addon structure",
+		Long:  "Create an addon scaffold along with the common files based on a Helm Chart for quick starting.",
+		Example: `	vela addon create mongodb --helm-repo-url=https://marketplace.azurecr.io/helm/v1/repo --chart=mongodb --version=12.1.16
+will create something like this:
+	mongodb/
+	├── definitions
+	├── metadata.yaml
+	├── readme.md
+	├── resources
+	│   └── mongodb.cue
+	├── schemas
+	└── template.yaml
+
+If you want to store the scaffold in a different directory, you can use the -p/--path flag:
+	vela addon create mongodb -p ./some/repo --helm-repo-url=https://marketplace.azurecr.io/helm/v1/repo --chart=mongodb --version=12.1.16
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return fmt.Errorf("an addon name is required")
+			}
+
+			return pkgaddon.CreateAddonFromHelmChart(args[0], path, helmRepoURL, chartName, chartVersion)
+		},
+	}
+
+	cmd.Flags().StringVar(&helmRepoURL, "helm-repo-url", "", "URL that points to a Helm repo")
+	err := cmd.MarkFlagRequired("helm-repo-url")
```

Comment:
why these parameters are required? Is it possible for other kinds of type except helm? \cc @wangyikewxgm @charlie0129 

What should the `vela addon create/init` happen if we don't specify any helm parameters?  We **MUST** allow this case, at least we can create an empty scaffold with same basic desciptions/folders. Please refer to what helm do.

Reply:
If that's the case, then there won't be much to be generated, or too complicated to describe using parameters. Maybe just use the `example` addon in the `experimental` registry. That's probably a good start.

Reply:
> Maybe just use the example addon in the experimental registry. 

Yes, that's somehow to the point. Make sure the first example can clarify all the concepts and rules of making an addon.

Reply:
> why these parameters are required? Is it possible for other kinds of type except helm? \cc @wangyikewxgm @charlie0129
> 
> What should the `vela addon create/init` happen if we don't specify any helm parameters? We **MUST** allow this case, at least we can create an empty scaffold with same basic desciptions/folders. Please refer to what helm do.

Helm parameter is optional. If without them will generate a basic structure.